04-onion-routing.md

```diff
@@ -1143,9 +1390,59 @@ The _origin node_:
   - MAY use the data specified in the various failure types for debugging
   purposes.
 
+# Onion Messages
+
+Onion messages allow peers to use existing connections to query for
+invoices (see [BOLT 12](12-offer-encoding.md)).  Like gossip messages,
+they are not associated with a particular local channel.  Like HTLCs,
+they use [onion messages](#onion-message-payload-format) protocol for
+end-to-end encryption.
+
+Onion messages are unreliable: in particular, they are designed to
+be cheap to process and require no storage to forward.  As a result,
+there is no error returned from intermediary nodes.
+
+For consistency, all onion messages use [Route Blinding](#route-blinding).
+
+## The `onion_message` Message
+
+1. type: 513 (`onion_message`) (`option_onion_messages`)
+2. data:
+    * [`point`:`blinding`]
+    * [`u16`:`len`]
+    * [`len*byte`:`onionmsg`]
+
+## Requirements
+
+The writer:
+- MUST populate the per-hop payloads as described in [Onion Message Payload Format](onion-message-payload-format).
+- SHOULD retry via a different route if it expects a response and
+  doesn't receive one after a reasonable period.
+- SHOULD set `len` to 1366 or 32834.
+
+The reader:
+- MUST handle the per-hop payloads as described in [Onion Message Payload Format](onion-message-payload-format).
+- SHOULD accept onion messages from peers without an established channel.
```

Comment:
I assume this is with Offers in mind but it is strange to me. Given the ongoing discussion about spam prevention and rate-limiting it seems reasonable to require at least a payment channel before accepting and forwarding onion messages

Reply:
You don't have to forward them, or you can ratelimit them, but you should definitely accept them.  Like you do pings...

Reply:
Noted in rationale.